### PR TITLE
#1068 - Validation of Avahi cluster name

### PR DESCRIPTION
**Pull request by kvanhijf**

### CODE REVIEW

**Review by JeffreyDevloo — APPROVED** [review]

File: ovs/lib/nodeinstallation.py [hunk 1]
```diff
@@ -216,14 +215,19 @@ def setup_node(node_type=None, execute_rollback=False):
                     if cluster_name == new_cluster:  # Create a new OVS cluster
                         first_node = True
                         while True:
+                            master_ip = Interactive.ask_choice(NodeInstallationController.host_ips, 'Select the public IP address of {0}'.format(node_name))
```
Comment (pploegaert) [inline ovs/lib/nodeinstallation.py:218]: This changes the order of the interactive setup ... pls move the question for the public IP after the cluster_name, as we now can ask to create a new cluster, get a question for the ip and then the question for the cluster_name

This breaks the interactive setup and requires an update of the documentation too ...

**Comment by wimpers** [comment]
@pploegaert  @kvanhijf  

what exactly needs to be changed in the documentation?

**Comment by pploegaert** [comment]
preferably nothing, just reorder the questions in the interactive setup as it was before
otherwise we need to update the nightly build setup too ...

**Comment by kvanhijf** [comment]
@wimpers : We now ask
"Select the public IP address" before "Please enter the cluster name"
This used to be the other way around

I'd rather not revert this, because of my previous explanation.
Either the framework changes the order in which the questions are asked or QA does it.
On the other hand i wonder why QA doesn't use the preconfig.json file for setting up environments like on 16.04, then we wouldn't have these issues. The manual installations are still being verified by the framework team anyway

**Comment by pploegaert** [comment]
@kvanhijf we test both automatic and manual,
when the interactive part fails we can at least check if this is valid or we at least need to update the tests and the documentation
this is more work so what does this change in order bring us?
this is not related to an avahi change anyway, right?

**Comment by kvanhijf** [comment]
It is related to the Avahi cluster name validation, because the IP is part of the internal Avahi cluster name, so if you change IP, the cluster name can become shorter/longer
Also imo, QA should reflect changes made by the framework in their code, the framework should not write code in order to let the tests written by QA succeed

**Comment by wimpers** [comment]
@kvanhijf has 2 points

**Pull request merged**